src/Microsoft.SqlTools.ServiceLayer/Scripting/Contracts/ScriptingErrorEvent.cs

```diff
@@ -0,0 +1,28 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using Microsoft.SqlTools.Hosting.Protocol.Contracts;
+
+namespace Microsoft.SqlTools.ServiceLayer.Scripting.Contracts
+{
+    /// <summary>
+    /// Parameters to be sent to indicate a scripting operations has encountered and error.
+    /// </summary>
+    public class ScriptingErrorParams : ScriptingEventParams
+    {
+        public string Message { get; set; }
+
+        public string DiagnosticMessage { get; set; }
+    }
+
+    /// <summary>
+    /// Event sent to indicate a scripting database operation has encountered an error.
+    /// </summary>
+    public class ScriptingErrorEvent
```

Comment:
Please check with Ben, this is a nice practice / looks good but I know he's been doing work in the error handling area and it'd be great if we align this so it's consistent.